core/ecschema-metadata/src/UnitConversion/UnitTree.ts

```diff
@@ -138,6 +196,29 @@ export class UnitGraph {
       .finally(() => this._unitsInProgress.delete(unit.key.fullName));
   }
 
+  /**
+   * Adds unit and corresponding children to graph as well as edges between units
+   * @param unit Current unit to be added to graph
+   */
+  public addUnitSync(unit: Unit | Constant): void {
+    // Check if unit is already being processed - but we can't wait for it in sync version
+    if(this._unitsInProgress.has(unit.key.fullName)) {
+      // In sync version, we either skip or throw an error since we can't await
+      throw new BentleyError(BentleyStatus.ERROR, "Unit is currently being processed asynchronously", () => {
+        return { unit: unit.key.fullName };
+      });
+    }
+
+    if (this._graph.hasNode(unit.key.fullName))
+      return;
+
+    this._graph.setNode(unit.key.fullName, unit);
+    if (this.isIdentity(unit))
+      return;
+
+    this.addUnitToGraphSync(unit);
```

Comment:
while everything looks fine on paper, it seems that both sync and async methods of the unitConverter will use the same `_graph` property. Do we have to worry about race conditions...

This is probably very minor, and seems like the graph acts as a cache in this sense, so the consequence might just be a very tiny effect of no cache hit